### PR TITLE
samples: mmss: fix connection initialization order

### PR DESCRIPTION
**Pull request by glarsennordic**
Slightly alter initialization procedure in connection.c to make sure delta firmware FOTA succeeds

IRIS4777

Signed-off-by: Georges Oates_Larsen <georges.larsen@nordicsemi.no>

### CODE REVIEW

File: samples/nrf9160/nrf_cloud_mqtt_multi_service/src/connection.c [hunk 1]
```diff
@@ -759,6 +770,20 @@ void connection_management_thread_fn(void)
...
+	 *  connection is lost).
+	 */
+	LOG_INF("Setting up LTE...");
+	if (setup_lte()) {
+		LOG_ERR("Fatal: LTE setup failed");
+		long_led_pattern(LED_FAILURE);
+		return;
+	}
+
+
```
Comment (plskeggs) [inline samples/nrf9160/nrf_cloud_mqtt_multi_service/src/connection.c:786]: remove extra blank line

**Review by plskeggs — APPROVED** [review]
Generally fine.

Commit message should be:
Jira: IRIS-4777
or
IRIS-4777

not IRIS4777

**Comment by jayteemo** [comment]
Adding DNM.  I’ll add my review comments tomorrow.

File: samples/nrf9160/nrf_cloud_mqtt_multi_service/src/connection.c [hunk 2]
```diff
@@ -642,7 +642,7 @@ static int connect_cloud(void)
  *
  * @return int - 0 on success, otherwise a negative error code.
  */
-static int setup_lte(void)
+static int setup_modem(void)
```
Comment (jayteemo) [inline samples/nrf9160/nrf_cloud_mqtt_multi_service/src/connection.c:645]: if FMFU is enabled and a delta update is performed, the code below will print out an invalid error message.
i think changing the check to `if (err < 0)` should be sufficient.
For reference, see https://github.com/nrfconnect/sdk-nrf/blob/052c17e36db62b5093538d0dc1be8ed0dab475f3/applications/asset_tracker_v2/src/main.c#L235

Comment (glarsennordic) [reply samples/nrf9160/nrf_cloud_mqtt_multi_service/src/connection.c:645]: Good catch -- I've decided to add debug logging for all three possibilities

File: samples/nrf9160/nrf_cloud_mqtt_multi_service/src/connection.c [hunk 3]
```diff
@@ -638,25 +638,70 @@ static int connect_cloud(void)
...
-		err = nrf_modem_lib_init(NORMAL_MODE);
-		if (err) {
-			LOG_ERR("Modem library initialization failed, error: %d", err);
+		ret = nrf_modem_lib_init(NORMAL_MODE);
+		if (ret < 0) {
+			LOG_ERR("Modem library initialization failed, error: %d", ret);
+		}
+		else if (ret > 0) {
+			LOG_DBG("Modem library completed firmware update and initialized with "
+			        "success code: %d", ret);
```
Comment (jayteemo) [inline samples/nrf9160/nrf_cloud_mqtt_multi_service/src/connection.c:657]: not all of the positive return values are "success" codes.

File: samples/nrf9160/nrf_cloud_mqtt_multi_service/src/connection.c [hunk 4]
```diff
@@ -638,25 +638,77 @@ static int connect_cloud(void)
...
 {
-	int err;
-
 	/* Initialize the modem library if required */
 	if (!IS_ENABLED(CONFIG_NRF_MODEM_LIB_SYS_INIT)) {
-		err = nrf_modem_lib_init(NORMAL_MODE);
-		if (err) {
-			LOG_ERR("Modem library initialization failed, error: %d", err);
+		/*
+		 * If there is a pending modem firmware update stored, nrf_modem_lib_init will
```
Comment (jayteemo) [inline samples/nrf9160/nrf_cloud_mqtt_multi_service/src/connection.c:650]:
```suggestion
		 * If there is a pending modem delta firmware update stored, nrf_modem_lib_init will
```

**Review by jayteemo — APPROVED** [review]

File: doc/nrf/releases/release-notes-changelog.rst [hunk 5]
```diff
@@ -437,6 +437,10 @@ nRF9160 samples
     * :ref:`lib_nrf_cloud` library is no longer de-initialized and re-initialized on disconnect and reconnect.
     * The :ref:`lib_nrf_cloud` library function :c:func:`nrf_cloud_gnss_msg_json_encode` is used to send PVT location data instead of building an NMEA sentence.
 
+  * Fixed:
+
+    * An issue with connection initialization causing delta modem FOTA updates to hang and require manual reset
```
Comment (peknis) [inline doc/nrf/releases/release-notes-changelog.rst:442]:
```suggestion
    * An issue with connection initialization causing delta modem FOTA updates to hang and require manual reset.
```

**Review by peknis — CHANGES_REQUESTED** [review]

**Comment by wiba-nordic** [comment]
The changelog entry should be added to [PR#8510](https://github.com/nrfconnect/sdk-nrf/pull/8510) as a comment at this point to prevent merge conflicts.

**Comment by greg-fer** [comment]
@wiba-nordic , @glarsennordic , I'm fine with the changelog entry in this PR -- easier to track down afterwards what change was due to which PR. I'll rebase my PR accordingly.

**Pull request merged**